packages/vite/src/node/logger.ts

```diff
@@ -164,6 +164,13 @@ export function printCommonServerUrls(
   }
 }
 
+const loopbackHosts = new Set<string | undefined>([
+  'localhost',
+  '127.0.0.1',
+  '::1',
+  '0000:0000:0000:0000:0000:0000:0000:0001'
+])
```

Comment:
I think we should use `new Set` without the `string | undefined` part, and update line 183 as 
```
if (hostname.host && loopbackHosts.has(hostname.host)) {
```

as `loopbackHosts` technically doesn't contain `undefined`